src/win/util.c

```diff
@@ -1283,3 +1283,101 @@ int uv_os_tmpdir(char* buffer, size_t* size) {
+                                                                              \
+    /* Allocate the destination buffer */                                     \
+    dst = uv__malloc(bufsize);                                                \
+                                                                              \
+    if (dst == NULL) {                                                        \
+      uv_free_passwd(pwd);                                                    \
+      return UV_ENOMEM;                                                       \
+    }                                                                         \
+                                                                              \
+    /* Convert to UTF-8 */                                                    \
+    bufsize = WideCharToMultiByte(CP_UTF8,                                    \
+                                  0,                                          \
+                                  src,                                        \
+                                  -1,                                         \
+                                  dst,                                        \
+                                  bufsize,                                    \
+                                  NULL,                                       \
+                                  NULL);                                      \
+                                                                              \
+    if (bufsize == 0) {                                                       \
+      uv_free_passwd(pwd);                                                    \
+      return uv_translate_sys_error(GetLastError());                          \
+    }                                                                         \
+  }                                                                           \
+  while (0)
+
+  if (pwd == NULL)
+    return UV_EINVAL;
+
+  /* Get the home directory using GetUserProfileDirectoryW() */
+  if (OpenProcessToken(GetCurrentProcess(), TOKEN_READ, &token) == 0)
+    return uv_translate_sys_error(GetLastError());
+
+  bufsize = MAX_PATH;
+  if (!GetUserProfileDirectoryW(token, path, &bufsize)) {
+    r = GetLastError();
+    CloseHandle(token);
+
+    /* This should not be possible */
+    if (r == ERROR_INSUFFICIENT_BUFFER)
+      return UV_EIO;
+
+    return uv_translate_sys_error(r);
+  }
+
+  CloseHandle(token);
+  CONVERT_BUFFER(path, pwd->homedir);
+
+  /* Get the username using GetUserNameW() */
+  bufsize = UNLEN + 1;
+  if (!GetUserNameW(username, &bufsize)) {
+    r = GetLastError();
+    uv_free_passwd(pwd);
+
+    /* This should not be possible */
+    if (r == ERROR_INSUFFICIENT_BUFFER)
+      return UV_EIO;
+
+    return uv_translate_sys_error(r);
+  }
+
+  CONVERT_BUFFER(username, pwd->name);
+#undef CONVERT_BUFFER
+
+  pwd->shell = NULL;
+  pwd->uid = -1;
```

Comment:
It might be necessary to define `uv_passwd_t` differently on Windows. `uv_uid_t` is currently defined as an unsigned character on Windows, so this shows up as 255.